packages/compass-collection/src/components/collection-tab.tsx

```diff
@@ -242,10 +248,38 @@ const CollectionTab = ({
   }
 
   return (
-    <CollectionTabWithMetadata
-      collectionMetadata={collectionMetadata}
-      {...props}
-    ></CollectionTabWithMetadata>
+    <>
+      <ElectronSubMenu label="&Collection">
+        <ElectronSubMenu label={collectionMetadata.namespace}>
+          <ElectronMenuItem
+            label="&Share Schema as JSON"
+            accelerator="Alt+CmdOrCtrl+S"
+            onClick={() => {
+              globalAppRegistry.emit('menu-share-schema-json');
+            }}
+          ></ElectronMenuItem>
+          <ElectronMenuSeparator></ElectronMenuSeparator>
+          {!collectionMetadata.isReadonly && (
+            <ElectronMenuItem
+              label="&Import Data"
+              onClick={() => {
+                globalAppRegistry.emit('open-active-namespace-import');
```

Comment:
This still uses these special case `active-namespace` events because it was the fastest way to make it work, but as we now render these contextually, we can completely drop those types of events from the code as all the context is available here